_posts/2016-10-21-puppet4-integration.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: Katello Puppet 4 integration
+date: 2016-10-19
+author: Zach Huntington-Meath
+tags:
+- katello
+- puppet
+excerpt: |
+    Puppet 4 integration is coming to Katello and we wanted to share with you how it works!
+---
+
+### Puppet 4 Integration with Katello
+
+Recently, Foreman and Puppet 4 became compatible. Now, Katello is ready to work with Puppet 4. It is available in Katello nightly and will be released in version 3.2.  In this post, we will explain how to create new Katello instances that are Puppet 4 ready, and upgrade existing systems.
+
+There are some important things to know about using Katello with Puppet 4:
+
+- Puppet 4 uses the Java Virtual Machine and requires more memory than Puppet 3 did. This means that systems acting as Katello servers with Puppet 4 need a larger amount of memory. At a minimum, we are requiring 8 GB of memory, and recommending 12GB
+
+- Puppet 4 masters can still provision hosts with Puppet 3 with [some configuration](https://docs.puppet.com/puppetserver/latest/compatibility_with_puppet_agent.html). This can help with systems that are migrating over to Puppet 4. You can start with upgrading your masters to Puppet 4 and then gradually upgrade your agents.
+
+- You can use this matrix as a quick reference of what can work together:
+
+| | Puppet 3 client | Puppet 4 client
+| :--- | :---: | :---: 
+| **Puppet 4 server** | Compatible | Compatible
+| **Puppet 3 server** | Compatible | Not compatible
```

Comment:
The table looks a little funny on the blog post, is there some setting we could set so the Markdown renders table borders?

![table](https://cloud.githubusercontent.com/assets/429763/19772812/65a4cd82-9c35-11e6-826c-b06b85f5427f.png)